frontend/include/chpl/resolution/resolution-types.h

```diff
@@ -1862,6 +1862,8 @@ class CallResolutionResult {
   MostSpecificCandidates mostSpecific_;
   // what is the type of the call expression?
   types::QualifiedType exprType_;
+  // if the called expression is an iterator, what is the yielded type?
+  types::QualifiedType yieldedType_;
```

Comment:
What is `exprType_` in such a case? Could we store the yielded type in `exprType_`?

Reply:
In such a case, `exprType_` will be `IteratorType`, which I think is necessary (since the `IteratorType` can be passed to other functions that accept `_iteratorRecord`, for example).